pkg/cmd/builder/build.go

```diff
@@ -192,69 +203,85 @@ func loadImage(ctx context.Context, in io.Reader, namespace, address, snapshotte
 	return nil
 }
 
-func generateBuildctlArgs(ctx context.Context, client *containerd.Client, options types.BuilderBuildOptions) (buildCtlBinary string,
-	buildctlArgs []string, needsLoading bool, metaFile string, tags []string, cleanup func(), err error) {
+type BuildctlArgsResult struct {
+	BuildctlArgs   []string
+	BuildctlBinary string
+	Cleanup        func()
+	DestFile       string
+	MetaFile       string
+	NeedsLoading   bool // Specifies whether the image needs to be loaded into the containerd image store
+	Tags           []string
+}
 
+func generateBuildctlArgs(ctx context.Context, client *containerd.Client, options types.BuilderBuildOptions) (result BuildctlArgsResult, err error) {
 	buildctlBinary, err := buildkitutil.BuildctlBinary()
 	if err != nil {
-		return "", nil, false, "", nil, nil, err
+		return result, err
 	}
+	result.BuildctlBinary = buildctlBinary
 
 	output := options.Output
 	if output == "" {
 		info, err := client.Server(ctx)
 		if err != nil {
-			return "", nil, false, "", nil, nil, err
+			return result, err
 		}
 		sharable, err := isImageSharable(options.BuildKitHost, options.GOptions.Namespace, info.UUID, options.GOptions.Snapshotter, options.Platform)
 		if err != nil {
-			return "", nil, false, "", nil, nil, err
+			return result, err
 		}
 		if sharable {
 			output = "type=image,unpack=true" // ensure the target stage is unlazied (needed for any snapshotters)
 		} else {
-			output = "type=docker"
+			// https://github.com/moby/buildkit?tab=readme-ov-file#output
+			// type=image is the native type for containerd
+			output = "type=image"
```

Comment:
Default to `type=image` instead of `type=docker`. See https://github.com/containerd/nerdctl/pull/3604#issue-2619462937